src/Gui/View3DInventor.cpp

```diff
@@ -773,6 +777,93 @@ void View3DInventor::setCurrentViewMode(ViewMode newmode)
     }
 }
 
+RayPickInfo View3DInventor::getObjInfoRay(Base::Vector3d* startvec,
+                                             Base::Vector3d* dirvec)
+{
+    double vsx,vsy,vsz;
+    double vdx,vdy,vdz;
+    vsx = startvec->x;
+    vsy = startvec->y;
+    vsz = startvec->z;
+    vdx = dirvec->x;
+    vdy = dirvec->y;
+    vdz = dirvec->z;
+
+    RayPickInfo ret = {
+        .isValid = false,
+        .point = Base::Vector3d(),
+        .document = "",
+        .object= "",
+        .parentObject = std::nullopt,
+        .component = std::nullopt,
+        .subName = std::nullopt
+    };
+    SoRayPickAction action(getViewer()->getSoRenderManager()->getViewportRegion());
+    action.setRay(SbVec3f(vsx, vsy, vsz), SbVec3f(vdx, vdy, vdz));
+    action.apply(getViewer()->getSoRenderManager()->getSceneGraph());
+    SoPickedPoint *Point = action.getPickedPoint();
+
+    if (Point) {

```

Comment:
This could use early exist:
```
if (!Point) {
    return ret;
}
```

Reply:
Thanks, formatting will be fixed in the next commit.